### PR TITLE
error if non-standard periodic freq without imu sampling spec

### PR DESCRIPTION
**Pull request by EwoudSmeur**
Warn the user so they don't forget to check IMU settings when changing the PERIODIC_FREQUENCY.

Example of old situation:
Changing the PERIODIC_FREQUENCY to 200 Hz with mpu9250 led to IMU being configured at 100 Hz, which completely messed up integration of the rates without any warning.

### CODE REVIEW

File: sw/airborne/modules/imu/imu_mpu9250_i2c.c [hunk 1]
```diff
@@ -57,6 +57,7 @@ PRINT_CONFIG_MSG("Gyro/Accel output rate is 2kHz at 8kHz internal sampling")
 #define IMU_MPU9250_GYRO_LOWPASS_FILTER MPU9250_DLPF_GYRO_41HZ
 #define IMU_MPU9250_ACCEL_LOWPASS_FILTER MPU9250_DLPF_ACCEL_41HZ
 PRINT_CONFIG_MSG("Gyro/Accel output rate is 100Hz at 1kHz internal sampling")
+#error Non-default PERIODIC_FREQUENCY: please define IMU_MPU9250_GYRO_LOWPASS_FILTER, IMU_MPU9250_ACCEL_LOWPASS_FILTER and IMU_MPU9250_SMPLRT_DIV.
```
Comment (gautierhattenberger) [inline sw/airborne/modules/imu/imu_mpu9250_i2c.c:60]: If it is an error, is it even necessary to define filter values ?

Comment (EwoudSmeur) [reply sw/airborne/modules/imu/imu_mpu9250_i2c.c:60]: I think I noticed an #error message doesn't stop you from finishing the compilation, so like this you could notice during compilation but you can also ignore it and flash... Not sure what is the desired behavior here, I would be fine with either.

Comment (dewagter) [reply sw/airborne/modules/imu/imu_mpu9250_i2c.c:60]: #error stops the compilation. Use a #warning instead if you want to still compile.

Comment (dewagter) [inline sw/airborne/modules/imu/imu_mpu9250_i2c.c:60]:
```suggestion
#warning Non-default PERIODIC_FREQUENCY: please define IMU_MPU9250_GYRO_LOWPASS_FILTER, IMU_MPU9250_ACCEL_LOWPASS_FILTER and IMU_MPU9250_SMPLRT_DIV.
```

File: sw/airborne/modules/imu/imu_mpu9250_spi.c [hunk 2]
```diff
@@ -63,6 +63,7 @@ PRINT_CONFIG_MSG("Gyro/Accel output rate is 2kHz at 8kHz internal sampling")
 #define IMU_MPU9250_GYRO_LOWPASS_FILTER MPU9250_DLPF_GYRO_41HZ
 #define IMU_MPU9250_ACCEL_LOWPASS_FILTER MPU9250_DLPF_ACCEL_41HZ
 PRINT_CONFIG_MSG("Gyro/Accel output rate is 100Hz at 1kHz internal sampling")
+#error Non-default PERIODIC_FREQUENCY: please define IMU_MPU9250_GYRO_LOWPASS_FILTER, IMU_MPU9250_ACCEL_LOWPASS_FILTER and IMU_MPU9250_SMPLRT_DIV.
```
Comment (dewagter) [inline sw/airborne/modules/imu/imu_mpu9250_spi.c:66]:
```suggestion
#warning Non-default PERIODIC_FREQUENCY: please define IMU_MPU9250_GYRO_LOWPASS_FILTER, IMU_MPU9250_ACCEL_LOWPASS_FILTER and IMU_MPU9250_SMPLRT_DIV.
```

File: sw/airborne/modules/imu/imu_mpu6000.c [hunk 3]
```diff
@@ -68,7 +68,7 @@ PRINT_CONFIG_MSG("Gyro/Accel output rate is 100Hz at 1kHz internal sampling")
 #ifndef IMU_MPU_ACCEL_LOWPASS_FILTER
 #define IMU_MPU_ACCEL_LOWPASS_FILTER MPU60X0_DLPF_ACC_44HZ // for ICM sensors
 #endif
-INFO("Non-default PERIODIC_FREQUENCY: using default IMU_MPU_LOWPASS_FILTER and IMU_MPU_SMPLRT_DIV.")
+#error Non-default PERIODIC_FREQUENCY: please define IMU_MPU_LOWPASS_FILTER and IMU_MPU_SMPLRT_DIV.
```
Comment (dewagter) [inline sw/airborne/modules/imu/imu_mpu6000.c:71]:
```suggestion
#warning Non-default PERIODIC_FREQUENCY: please define IMU_MPU_LOWPASS_FILTER and IMU_MPU_SMPLRT_DIV.
```

**Review by dewagter — APPROVED** [review]
LGTM

**Pull request merged**